cmd/avalanche-network-runner/control/control.go

```diff
@@ -411,20 +415,20 @@ func addNodeFunc(cmd *cobra.Command, args []string) error {
 	}
 	defer cli.Close()
 
-	if nodeConfig != "" {
-		color.Outf("{{yellow}}WARNING: overriding node configs with custom provided config {{/}} %+v\n", nodeConfig)
+	opts := []client.OpOption{
+		client.WithLogLevel(logLevel),
+	}
+
+	if addNodeConfig != "" {
+		color.Outf("{{yellow}}WARNING: overriding node configs with custom provided config {{/}} %+v\n", addNodeConfig)
 		// validate it's valid JSON
 		var js json.RawMessage
-		if err := json.Unmarshal([]byte(nodeConfig), &js); err != nil {
+		if err := json.Unmarshal([]byte(addNodeConfig), &js); err != nil {
 			return fmt.Errorf("failed to validate JSON for provided config file: %s", err)
 		}
+		opts = append(opts, client.WithGlobalNodeConfig(addNodeConfig))
```

Comment:
Is this supposed to be `WithGlobalNodeConfig`? This isn't a global config, and I don't think we do anything with the `globalNodeConfig` option in `AddNode`

Reply:
Ok so AddNode contains `StartRequest` as its gRPC data container because there are common params. In `StartRequest` we only have `GlobalConfig` though, no individual node-config entry.

On the server side, `AddNode` indeed *does* read `req.StartRequest.GetGlobalNodeConfig()` to then evaluate the config, the same way it does when receiving an actual `StartRequest`.

I don't think that's too bad. However, we could add a `NodeConfig` entry to the `AddNodeRequest` container if you insist, and use that instead also on server side.